[2018-04-18 03:23:27,604] WARN Client session timed out, have not heard from server in 1523883ms for sessionid 0x1001d761f0e002a (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:28,782] INFO Client session timed out, have not heard from server in 1523883ms for sessionid 0x1001d761f0e002a, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:28,327] WARN Client session timed out, have not heard from server in 1524943ms for sessionid 0x1001d761f0e002b (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:28,787] INFO Client session timed out, have not heard from server in 1524943ms for sessionid 0x1001d761f0e002b, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:28,577] INFO [Partition wing-binlog-event-0 broker=1] Shrinking ISR from 0,2,1 to 1 (kafka.cluster.Partition)
[2018-04-18 03:23:28,786] INFO [Partition wing-binlog-event-2 broker=0] Shrinking ISR from 0,2,1 to 0 (kafka.cluster.Partition)
[2018-04-18 03:23:28,887] INFO [ZooKeeperClient] Waiting until connected. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:28,887] INFO [ZooKeeperClient] Connected. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:28,894] INFO [ZooKeeperClient] Waiting until connected. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:28,894] INFO [ZooKeeperClient] Connected. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:29,105] INFO Opening socket connection to server 127.0.0.1/127.0.0.1:12183. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,106] INFO Socket connection established to 127.0.0.1/127.0.0.1:12183, initiating session (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,214] INFO Unable to read additional data from server sessionid 0x1001d761f0e002a, likely server has closed socket, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,321] INFO [ZooKeeperClient] Waiting until connected. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:29,566] INFO Opening socket connection to server 127.0.0.1/127.0.0.1:12183. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,567] INFO Socket connection established to 127.0.0.1/127.0.0.1:12183, initiating session (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,573] INFO Unable to read additional data from server sessionid 0x1001d761f0e002b, likely server has closed socket, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,616] INFO Opening socket connection to server 127.0.0.1/127.0.0.1:12182. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,617] INFO Socket connection established to 127.0.0.1/127.0.0.1:12182, initiating session (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,622] WARN Unable to reconnect to ZooKeeper service, session 0x1001d761f0e002a has expired (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,622] INFO Unable to reconnect to ZooKeeper service, session 0x1001d761f0e002a has expired, closing socket connection (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,623] INFO EventThread shut down for session: 0x1001d761f0e002a (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,624] INFO [ZooKeeperClient] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:29,642] ERROR Uncaught exception in scheduled task 'isr-expiration' (kafka.utils.KafkaScheduler)
kafka.zookeeper.ZooKeeperClientExpiredException: Session expired either before or while waiting for connection
	at kafka.zookeeper.ZooKeeperClient.$anonfun$waitUntilConnected$3(ZooKeeperClient.scala:233)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at kafka.utils.CoreUtils$.inLock(CoreUtils.scala:250)
	at kafka.zookeeper.ZooKeeperClient.waitUntilConnected(ZooKeeperClient.scala:221)
	at kafka.zookeeper.ZooKeeperClient.$anonfun$waitUntilConnected$1(ZooKeeperClient.scala:215)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at kafka.utils.CoreUtils$.inLock(CoreUtils.scala:250)
	at kafka.zookeeper.ZooKeeperClient.waitUntilConnected(ZooKeeperClient.scala:215)
	at kafka.zk.KafkaZkClient.retryRequestsUntilConnected(KafkaZkClient.scala:1461)
	at kafka.zk.KafkaZkClient.kafka$zk$KafkaZkClient$$retryRequestUntilConnected(KafkaZkClient.scala:1437)
	at kafka.zk.KafkaZkClient.conditionalUpdatePath(KafkaZkClient.scala:595)
	at kafka.utils.ReplicationUtils$.updateLeaderAndIsr(ReplicationUtils.scala:34)
	at kafka.cluster.Partition.updateIsr(Partition.scala:670)
	at kafka.cluster.Partition.$anonfun$maybeShrinkIsr$1(Partition.scala:513)
	at scala.runtime.java8.JFunction0$mcZ$sp.apply(JFunction0$mcZ$sp.java:12)
	at kafka.utils.CoreUtils$.inLock(CoreUtils.scala:250)
	at kafka.utils.CoreUtils$.inWriteLock(CoreUtils.scala:258)
	at kafka.cluster.Partition.maybeShrinkIsr(Partition.scala:504)
	at kafka.server.ReplicaManager.$anonfun$maybeShrinkIsr$2(ReplicaManager.scala:1335)
	at kafka.server.ReplicaManager.$anonfun$maybeShrinkIsr$2$adapted(ReplicaManager.scala:1335)
	at scala.collection.Iterator.foreach(Iterator.scala:929)
	at scala.collection.Iterator.foreach$(Iterator.scala:929)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
	at kafka.server.ReplicaManager.maybeShrinkIsr(ReplicaManager.scala:1335)
	at kafka.server.ReplicaManager.$anonfun$startup$1(ReplicaManager.scala:322)
	at kafka.utils.KafkaScheduler.$anonfun$schedule$2(KafkaScheduler.scala:110)
	at kafka.utils.CoreUtils$$anon$1.run(CoreUtils.scala:62)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[2018-04-18 03:23:29,643] INFO [ZooKeeperClient] Initializing a new session to 127.0.0.1:12181,127.0.0.1:12182,127.0.0.1:12183. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:29,645] INFO Initiating client connection, connectString=127.0.0.1:12181,127.0.0.1:12182,127.0.0.1:12183 sessionTimeout=6000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@3c130745 (org.apache.zookeeper.ZooKeeper)
[2018-04-18 03:23:29,682] INFO [ZooKeeperClient] Waiting until connected. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:29,701] INFO Opening socket connection to server 127.0.0.1/127.0.0.1:12183. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,706] INFO Socket connection established to 127.0.0.1/127.0.0.1:12183, initiating session (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,714] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2018-04-18 03:23:29,721] INFO Unable to read additional data from server sessionid 0x0, likely server has closed socket, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,834] INFO [ZooKeeperClient] Waiting until connected. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:29,834] INFO [ZooKeeperClient] Waiting until connected. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:29,871] INFO Opening socket connection to server 127.0.0.1/127.0.0.1:12182. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,871] INFO Socket connection established to 127.0.0.1/127.0.0.1:12182, initiating session (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,875] WARN Unable to reconnect to ZooKeeper service, session 0x1001d761f0e002b has expired (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,875] INFO Unable to reconnect to ZooKeeper service, session 0x1001d761f0e002b has expired, closing socket connection (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,875] INFO EventThread shut down for session: 0x1001d761f0e002b (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,876] INFO [ZooKeeperClient] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:29,879] INFO [ZooKeeperClient] Initializing a new session to 127.0.0.1:12181,127.0.0.1:12182,127.0.0.1:12183. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:29,879] INFO Initiating client connection, connectString=127.0.0.1:12181,127.0.0.1:12182,127.0.0.1:12183 sessionTimeout=6000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@3c130745 (org.apache.zookeeper.ZooKeeper)
[2018-04-18 03:23:29,888] INFO Opening socket connection to server 127.0.0.1/127.0.0.1:12182. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,888] INFO Socket connection established to 127.0.0.1/127.0.0.1:12182, initiating session (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:29,890] ERROR Uncaught exception in scheduled task 'isr-expiration' (kafka.utils.KafkaScheduler)
kafka.zookeeper.ZooKeeperClientExpiredException: Session expired either before or while waiting for connection
	at kafka.zookeeper.ZooKeeperClient.$anonfun$waitUntilConnected$3(ZooKeeperClient.scala:233)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at kafka.utils.CoreUtils$.inLock(CoreUtils.scala:250)
	at kafka.zookeeper.ZooKeeperClient.waitUntilConnected(ZooKeeperClient.scala:221)
	at kafka.zookeeper.ZooKeeperClient.$anonfun$waitUntilConnected$1(ZooKeeperClient.scala:215)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at kafka.utils.CoreUtils$.inLock(CoreUtils.scala:250)
	at kafka.zookeeper.ZooKeeperClient.waitUntilConnected(ZooKeeperClient.scala:215)
	at kafka.zk.KafkaZkClient.retryRequestsUntilConnected(KafkaZkClient.scala:1461)
	at kafka.zk.KafkaZkClient.kafka$zk$KafkaZkClient$$retryRequestUntilConnected(KafkaZkClient.scala:1437)
	at kafka.zk.KafkaZkClient.conditionalUpdatePath(KafkaZkClient.scala:595)
	at kafka.utils.ReplicationUtils$.updateLeaderAndIsr(ReplicationUtils.scala:34)
	at kafka.cluster.Partition.updateIsr(Partition.scala:670)
	at kafka.cluster.Partition.$anonfun$maybeShrinkIsr$1(Partition.scala:513)
	at scala.runtime.java8.JFunction0$mcZ$sp.apply(JFunction0$mcZ$sp.java:12)
	at kafka.utils.CoreUtils$.inLock(CoreUtils.scala:250)
	at kafka.utils.CoreUtils$.inWriteLock(CoreUtils.scala:258)
	at kafka.cluster.Partition.maybeShrinkIsr(Partition.scala:504)
	at kafka.server.ReplicaManager.$anonfun$maybeShrinkIsr$2(ReplicaManager.scala:1335)
	at kafka.server.ReplicaManager.$anonfun$maybeShrinkIsr$2$adapted(ReplicaManager.scala:1335)
	at scala.collection.Iterator.foreach(Iterator.scala:929)
	at scala.collection.Iterator.foreach$(Iterator.scala:929)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
	at kafka.server.ReplicaManager.maybeShrinkIsr(ReplicaManager.scala:1335)
	at kafka.server.ReplicaManager.$anonfun$startup$1(ReplicaManager.scala:322)
	at kafka.utils.KafkaScheduler.$anonfun$schedule$2(KafkaScheduler.scala:110)
	at kafka.utils.CoreUtils$$anon$1.run(CoreUtils.scala:62)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[2018-04-18 03:23:29,890] INFO Creating /brokers/ids/1 (is it secure? false) (kafka.zk.KafkaZkClient)
[2018-04-18 03:23:30,515] INFO Opening socket connection to server 127.0.0.1/127.0.0.1:12182. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:30,515] INFO Socket connection established to 127.0.0.1/127.0.0.1:12182, initiating session (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:30,657] INFO Unable to read additional data from server sessionid 0x0, likely server has closed socket, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:30,661] INFO Unable to read additional data from server sessionid 0x0, likely server has closed socket, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:30,767] INFO [ZooKeeperClient] Waiting until connected. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:30,768] INFO [ZooKeeperClient] Waiting until connected. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:31,272] INFO Opening socket connection to server 127.0.0.1/127.0.0.1:12181. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:31,274] INFO Socket connection established to 127.0.0.1/127.0.0.1:12181, initiating session (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:31,310] INFO Session establishment complete on server 127.0.0.1/127.0.0.1:12181, sessionid = 0x1002434d1da0000, negotiated timeout = 6000 (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:31,310] INFO [ZooKeeperClient] Connected. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:31,310] INFO [ZooKeeperClient] Connected. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:31,368] ERROR Error while creating ephemeral at /brokers/ids/1, node already exists and owner '72089987202285611' does not match current session '72097403315552256' (kafka.zk.KafkaZkClient$CheckedEphemeral)
[2018-04-18 03:23:31,368] INFO Result of znode creation at /brokers/ids/1 is: NODEEXISTS (kafka.zk.KafkaZkClient)
[2018-04-18 03:23:31,492] INFO Opening socket connection to server 127.0.0.1/127.0.0.1:12181. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:31,493] INFO Socket connection established to 127.0.0.1/127.0.0.1:12181, initiating session (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:31,507] INFO Session establishment complete on server 127.0.0.1/127.0.0.1:12181, sessionid = 0x1002434d1da0001, negotiated timeout = 6000 (org.apache.zookeeper.ClientCnxn)
[2018-04-18 03:23:31,507] INFO [ZooKeeperClient] Connected. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:31,507] INFO [ZooKeeperClient] Connected. (kafka.zookeeper.ZooKeeperClient)
[2018-04-18 03:23:31,523] ERROR Error while creating ephemeral at /brokers/ids/0, node already exists and owner '72089987202285610' does not match current session '72097403315552257' (kafka.zk.KafkaZkClient$CheckedEphemeral)
[2018-04-18 03:23:31,523] INFO Result of znode creation at /brokers/ids/0 is: NODEEXISTS (kafka.zk.KafkaZkClient)
